### PR TITLE
[Qt] Refactor QMapbox Annotations API

### PR DESCRIPTION
**Pull request by brunoabinader**
Add new types in `QMapbox` namespace to support the new `geometry.hpp`-based geometries.

### CODE REVIEW

File: platform/qt/src/qmapboxgl.cpp [hunk 1]
```diff
@@ -848,42 +836,114 @@ void QMapboxGL::setTransitionOptions(qint64 duration, qint64 delay) {
     d_ptr->mapObj->setTransitionOptions(mbgl::style::TransitionOptions{ convert(duration), convert(delay) });
 }
 
-mbgl::Annotation fromPointAnnotation(const PointAnnotation &pointAnnotation) {
-    const Coordinate &coordinate = pointAnnotation.first;
-    const QString &icon = pointAnnotation.second;
-    return mbgl::SymbolAnnotation { mbgl::Point<double> { coordinate.second, coordinate.first }, icon.toStdString() };
-}
+mbgl::ShapeAnnotationGeometry asMapboxGLGeometry(const QMapbox::ShapeAnnotationGeometry &geometry) {
+    auto asMapboxGLLineString = [&](const QMapbox::Coordinates &lineString) {
```
Comment (tmpsantos) [inline platform/qt/src/qmapboxgl.cpp]: My suggestion here is to move these conversion functions to `platform/qt/src/qt_conversion.hpp` and rename them to `toSomething` which is the pattern we use there.

Comment (brunoabinader) [inline platform/qt/src/qmapboxgl.cpp]: @tmpsantos let's do that on a separate PR, which also involves moving e.g. `fromStringList` and `toSpriteImage` which do not belong to this change.
